Support rfr2 devices

The lines from this patch were copied from the
401-gcc-non-public-devices-support.patch from the Atmel toolchain 3.4.2.
These two devices were picked to match the backported upstream r2038 in
avr-libc (which also adds support for 64rfr2, but gcc already supports
that one).

--- gcc/config/avr/avr-mcus.def
+++ gcc/config/avr/avr-mcus.def
@@ -228,6 +228,7 @@ AVR_MCU ("atmega64m1",           ARCH_AVR5, "__AVR_ATmega64M1__",        0, 0, 0
 AVR_MCU ("atmega64hve",          ARCH_AVR5, "__AVR_ATmega64HVE__",       0, 0, 0x0100, 1, "m64hve")
 AVR_MCU ("atmega64rfa2",         ARCH_AVR5, "__AVR_ATmega64RFA2__",      0, 0, 0x0200, 1, "m64rfa2")
 AVR_MCU ("atmega64rfr2",         ARCH_AVR5, "__AVR_ATmega64RFR2__",      0, 0, 0x0200, 1, "m64rfr2")
+AVR_MCU ("atmega644rfr2",        ARCH_AVR5, "__AVR_ATmega644RFR2__",     0, 0, 0x0200, 1, "m644rfr2")
 AVR_MCU ("atmega32hvbrevb",      ARCH_AVR5, "__AVR_ATmega32HVBREVB__",   0, 0, 0x0100, 1, "m32hvbrevb")
 AVR_MCU ("atmega48hvf",          ARCH_AVR5, "__AVR_ATmega48HVF__",       0, 0, 0x0100, 1, "m48hvf")
 AVR_MCU ("at90can32",            ARCH_AVR5, "__AVR_AT90CAN32__",         0, 0, 0x0100, 1, "can32")
@@ -249,6 +250,8 @@ AVR_MCU ("atmega1281",           ARCH_AVR51, "__AVR_ATmega1281__",       0, 0, 0
 AVR_MCU ("atmega1284",           ARCH_AVR51, "__AVR_ATmega1284__",       0, 0, 0x0100, 2, "m1284")
 AVR_MCU ("atmega1284p",          ARCH_AVR51, "__AVR_ATmega1284P__",      0, 0, 0x0100, 2, "m1284p")
 AVR_MCU ("atmega128rfa1",        ARCH_AVR51, "__AVR_ATmega128RFA1__",    0, 0, 0x0200, 2, "m128rfa1")
+AVR_MCU ("atmega128rfr2",        ARCH_AVR51, "__AVR_ATmega128RFR2__",    0, 0, 0x0200, 2, "m128rfr2")
+AVR_MCU ("atmega1284rfr2",       ARCH_AVR51, "__AVR_ATmega1284RFR2__",   0, 0, 0x0200, 2, "m1284rfr2")
 AVR_MCU ("at90can128",           ARCH_AVR51, "__AVR_AT90CAN128__",       0, 0, 0x0100, 2, "can128")
 AVR_MCU ("at90usb1286",          ARCH_AVR51, "__AVR_AT90USB1286__",      0, 0, 0x0100, 2, "usb1286")
 AVR_MCU ("at90usb1287",          ARCH_AVR51, "__AVR_AT90USB1287__",      0, 0, 0x0100, 2, "usb1287")
@@ -256,6 +259,8 @@ AVR_MCU ("at90usb1287",          ARCH_AVR51, "__AVR_AT90USB1287__",      0, 0, 0
 AVR_MCU ("avr6",                 ARCH_AVR6, NULL,                        0, 0, 0x0200, 4, "m2561")
 AVR_MCU ("atmega2560",           ARCH_AVR6, "__AVR_ATmega2560__",        0, 0, 0x0200, 4, "m2560")
 AVR_MCU ("atmega2561",           ARCH_AVR6, "__AVR_ATmega2561__",        0, 0, 0x0200, 4, "m2561")
+AVR_MCU ("atmega256rfr2",        ARCH_AVR6, "__AVR_ATmega256RFR2__",     0, 0, 0x0200, 4, "m256rfr2")
+AVR_MCU ("atmega2564rfr2",       ARCH_AVR6, "__AVR_ATmega2564RFR2__",    0, 0, 0x0200, 4, "m2564rfr2")
 /* Xmega, 16K <= Flash < 64K, RAM <= 64K */
 AVR_MCU ("avrxmega2",        ARCH_AVRXMEGA2, NULL,                       0, 0, 0x2000, 1, "x32a4")
 AVR_MCU ("atxmega16a4",      ARCH_AVRXMEGA2, "__AVR_ATxmega16A4__",      0, 0, 0x2000, 1, "x16a4")
